common/src/main/java/io/netty/util/NettyRuntime.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util;
+
+import io.netty.util.internal.SystemPropertyUtil;
+
+import java.util.Locale;
+
+/**
+ * A utility class for wrapping calls to {@link Runtime}.
+ */
+public final class NettyRuntime {
+
+    /**
+     * Holder class for available processors to enable testing.
+     */
+    static class AvailableProcessorsHolder {
+
+        private final Object lock = new Object();
+
+        private int availableProcessors;
+
+        /**
+         * Set the number of available processors.
+         *
+         * @param availableProcessors the number of available processors
+         * @throws IllegalArgumentException if the specified number of available processors is non-positive
+         * @throws IllegalStateException    if the number of available processors is already configured
+         */
+        void setAvailableProcessors(final int availableProcessors) {
+            synchronized (lock) {
```

Comment:
Oops, artifact of an earlier iteration that I never refactored into the simpler form after what I had done was no longer needed. I pushed feea27a5735c12d1aa99ec96a8c9bc1f9ddfa86c.